docs/book/v3/quick-start.md

```diff
@@ -1,99 +1,166 @@
-objects, as well as extracting data sets from them.
+The component contains [concrete implementations](#available_implementations) for a number of common use cases, such as by using arrays, object methods, and reflection, and provides [interfaces](#base_interfaces) for creating custom implementations.
 
-The component consists of interfaces, and several implementations for common use cases.
+## Basic Usage
 
-## Base Interfaces
+### Hydrating an Object
 
-### ExtractionInterface
+To hydrate an object with data, instantiate the hydrator and then pass to it the data for hydrating the object.
 
 ```php
-namespace Laminas\Hydrator;
+use Laminas\Hydrator;
 
-interface ExtractionInterface
-{
-    /**
-     * Extract values from an object
-     *
-     * @return mixed[]
-     */
-    public function extract(object $object) : array;
-}
+$hydrator = new Hydrator\ArraySerializableHydrator();
+
+$data = [
+    'first_name'    => 'James',
+    'last_name'     => 'Kahn',
+    'email_address' => 'james.kahn@example.org',
+    'phone_number'  => '+61 419 1234 5678',
+];
+
+$object = $hydrator->hydrate($data, new ArrayObject());
 ```
 
-### HydrationInterface
+### Extracting Values From an Object
+
+To extract data from an object, instantiate the applicable hydrator and then call `extract`, passing to it the object to extract data from.
 
 ```php
-namespace Laminas\Hydrator;
+use Laminas\Hydrator;
 
-interface HydrationInterface
-{
-    /**
-     * Hydrate $object with the provided $data.
-     *
-     * @param mixed[] $data
-     * @return object The implementation should return an object of any type.
-     *     By purposely omitting the return type from the signature,
-     *     implementations may choose to specify a more specific type.
-     */
-    public function hydrate(array $data, object $object);
-}
+$hydrator = new Hydrator\ArraySerializableHydrator();
+
+// ... Assuming that $object has already been initialised
+$data = $hydrator->extract($object);
 ```
 
-### HydratorInterface
+## Available Implementations
+
+### ArraySerializableHydrator
+
+The ArraySerializableHydrator hydrates data from an array and extracts an object’s data returning it as an array.
+Objects passed to the hydrate method must implement either `exchangeArray()` or `populate()` to support hydration, and must implement `getArrayCopy()` to support extraction.
+
+### ClassMethodsHydrator
+
+The ClassMethodsHydrator calls "setter" methods matching keys in the data set to hydrate objects and calls "getter" methods matching keys in the data set during extraction, based on the following rules:
+
+- `is*()`, `has*()`, and `get*()` methods will be used when extracting data.
+  The method prefix will be removed from the key's name.
+- `set*()` methods will be used when hydrating properties.
 
 ```php
-namespace Laminas\Hydrator;
+<?php
 
-interface HydratorInterface extends
-    ExtractionInterface,
-    HydrationInterface
+use Laminas\Hydrator;
```

Comment:
The same like before, the namespace is used only once therefore it can be reduced on one line.